.github/workflows/tests-clustermesh-upgrade.yaml

```diff
@@ -424,23 +428,79 @@ jobs:
             ${{ steps.newest-vars.outputs.cilium_install_defaults }} \
             ${{ steps.vars.outputs.cilium_install_defaults }} \
             ${{ steps.kvstore.outputs.cilium_install_kvstore }} \
-            ${{ steps.clustermesh-vars.outputs.cilium_install_cluster1 }} \
-            --set clustermesh.apiserver.kvstoremesh.enabled=true
+            ${{ steps.clustermesh-vars.outputs.cilium_install_cluster1 }}
 
-      - name: Rollout Cilium agents in cluster2
-        if: ${{ !matrix.external-kvstore }}
+      - name: Wait for cluster mesh status to be ready
         run: |
-          # This makes sure that the remote agents reconnect to the new instance of the
-          # clustermesh-apiserver, without waiting for the watchdog mechanism to kick in.
-          kubectl --context ${{ env.contextName2 }} rollout restart -n kube-system ds/cilium
+          cilium --context ${{ env.contextName1 }} status --wait
+          cilium --context ${{ env.contextName2 }} status --wait
+          cilium --context ${{ env.contextName1 }} clustermesh status --wait --wait-duration=5m
+          cilium --context ${{ env.contextName2 }} clustermesh status --wait --wait-duration=5m
+
+      - name: Write the Service manifest for testing failover
+        run: |
+          cat << EOF > echo-failover.yaml
+          apiVersion: v1
+          kind: Service
+          metadata:
+            annotations:
+              service.cilium.io/global: "true"
+            labels:
+              kind: echo
+              context: failover
+            name: echo-other-node-failover
+            namespace: cilium-test
+          spec:
+            ipFamilies:
+            - IPv4
+            - IPv6
+            ipFamilyPolicy: PreferDualStack
+            ports:
+            - name: http
+              port: 80
+              protocol: TCP
+              targetPort: 8080
+            selector:
+              name: echo-other-node
+            sessionAffinity: None
+            type: ClusterIP
+          EOF
+
+      - name: Restart clustermesh-apiserver and ensure client can connect to new Service
+        run: |
+          echo "Restarting clustermesh-apiserver deployments"
+          kubectl --context ${{ env.contextName2 }} -n kube-system rollout restart deployment -l k8s-app=clustermesh-apiserver
+          kubectl --context ${{ env.contextName2 }} -n kube-system rollout status deployment -l k8s-app=clustermesh-apiserver
+
+          echo "Deploying a global Service to test failover"
+          kubectl --context ${{ env.contextName1 }} apply -f echo-failover.yaml
+          kubectl --context ${{ env.contextName2 }} apply -f echo-failover.yaml
+
+          echo "Testing client connection to global Service"
+          kubectl --context ${{ env.contextName1 }} -n cilium-test exec deploy/client -i -- curl -s -v --connect-timeout 2 --max-time 5 --retry-max-time 60 --retry-all-errors --retry 10 --fail echo-other-node-failover
```

Comment:
Nit. Could also add the `--output /dev/null` option to avoid displaying the retrieved page. Same below.